docs/cccl/development/macro.rst

```diff
@@ -49,30 +49,41 @@ the compiler's major version number. For example, when the compiler is ``gcc-9.1
 | ``_CCCL_CUDA_COMPILER(CLANG)`` | Clang                   |
 +--------------------------------+-------------------------+
 
-The ``_CCCL_CUDA_COMPILER`` function-like macro can also be used to check the version of a compiler.
+The ``_CCCL_CUDA_COMPILER`` function-like macro can also be used to check the version of a CUDA compiler.
 
 .. code:: cpp
 
    _CCCL_CUDA_COMPILER(NVCC, <, 12, 3)
    _CCCL_CUDA_COMPILER(CLANG, >=, 14)
 
+*Note*: ``_CCCL_CUDA_COMPILER(...)`` check may result in a ``true`` value even during the compilation of a C++ source
+file. Use ``_CCCL_CUDA_COMPILATION()`` to check for the compilation of a CUDA source file.
+
 **CUDA identification/version macros**:
 
-+----------------------------------+-----------------------------+
-| ``_CCCL_HAS_CUDA_COMPILER()``    | CUDA compiler is available  |
-+----------------------------------+-----------------------------+
-| ``_CCCL_CUDACC_BELOW(12, 7)``    | CUDA version below 12.7     |
-+----------------------------------+-----------------------------+
-| ``_CCCL_CUDACC_AT_LEAST(12, 7)`` | CUDA version at least 12.7  |
-+----------------------------------+-----------------------------+
++----------------------------------+---------------------------------------------------------------------------------------------------------+
+| ``_CCCL_HAS_CUDA_COMPILER()``    | CUDA compiler is available                                                                              |
++----------------------------------+---------------------------------------------------------------------------------------------------------+
+| ``_CCCL_CUDA_COMPILATION()``     | CUDA source file is being compiled (equivalent to ``defined(__CUDACC__)``)                              |
++----------------------------------+---------------------------------------------------------------------------------------------------------+
+| ``_CCCL_HOST_COMPILATION()``     | Compiling host code, ``true`` when executing the CUDA ``__host__`` pass or compiling a C++ source files |
++----------------------------------+---------------------------------------------------------------------------------------------------------+
+| ``_CCCL_DEVICE_COMPILATION()``   | Compiling device code, ``true`` when executing the CUDA ``__device__`` pass                             |
++----------------------------------+---------------------------------------------------------------------------------------------------------+
```

Comment:
I don't like how you're using the attributes here to specify which compiler pass we're dealing with. I'd much rather just use prose here.